atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -284,12 +286,15 @@ private void saveSweepResults(final SweepProgressRowResult progress,
                                   final SweepResults results) {
         final long cellsDeleted = fromNullable(progress.getCellsDeleted()) + results.getCellsDeleted();
         final long cellsExamined = fromNullable(progress.getCellsExamined()) + results.getCellsExamined();
+        final long minimumSweptTimestamp = progress.getMinimumSweptTimestamp() == null ?
+                results.getSweptTimestamp() :
+                Math.min(progress.getMinimumSweptTimestamp(), results.getSweptTimestamp());
```

Comment:
if there's no minimumSweptTimestamp, we don't know how long ago the previous sweep run was. So I think this sweep will just have to have a minimumSweptTimestamp of 0.